samza-core/src/main/java/org/apache/samza/storage/blobstore/util/DirDiffUtil.java

```diff
@@ -168,11 +171,17 @@ public static BiPredicate<File, FileIndex> areSameFile(boolean compareLargeFileC
 
           // Don't compare file timestamps. The ctime of a local file just restored will be different than the
           // remote file, and will cause the file to be uploaded again during the first commit after restore.
-          areSameFiles = localFileAttrs.size() == remoteFileMetadata.getSize() &&
-              groupCache.get(String.valueOf(Files.getAttribute(localFile.toPath(), "unix:gid")),
-                () -> localFileAttrs.group().getName()).equals(remoteFileMetadata.getGroup()) &&
-              ownerCache.get(String.valueOf(Files.getAttribute(localFile.toPath(), "unix:uid")),
-                () -> localFileAttrs.owner().getName()).equals(remoteFileMetadata.getOwner());
+          areSameFiles = localFileAttrs.size() == remoteFileMetadata.getSize();
```

Comment:
On a side note is this sufficient? what if the files are the same size but with different values?

Reply:
RocksDB provides a guarantee that sst files of same names are same files and if the file names are different, their contents are definitely different as well.
Given that, this is just a sanity check to ensure that the files are same. This is, for example, checked after restore to make sure all the chunks of the blobs are actually downloaded. BTW we always re-download non-sst files, so this is not an issue for them.

Reply:
Synced up offline, SST files are immutable as per [original RocksDB paper](https://scontent-sjc3-1.xx.fbcdn.net/v/t39.8562-6/260203686_676839503287444_5314493774276549632_n.pdf?_nc_cat=103&ccb=1-7&_nc_sid=ad8a9d&_nc_ohc=y92R003NFBMAX-5A9g8&_nc_ht=scontent-sjc3-1.xx&oh=00_AfAYFM8VJ8d_nyA0H3vlxKIQFDG-WXKGio_Nu2__r-Nevw&oe=64EACD7C). Since we already checked the name and the size, these checks are sufficient